dspace-api/src/main/java/org/dspace/administer/FileDownloaderConfiguration.java

```diff
@@ -35,6 +39,20 @@ public void setDspaceRunnableClass(Class<FileDownloader> dspaceRunnableClass) {
         this.dspaceRunnableClass = dspaceRunnableClass;
     }
 
+    /**
+     * This script is allowed to execute to any authorized user. Further access control mechanism then checks,
+     * if the current user is authorized to download a file to the item specified in command line parameters.
+     *
+     * @param context   The relevant DSpace context
+     * @param commandLineParameters the parameters that will be used to start the process if known,
+     *        <code>null</code> otherwise
+     * @return          A boolean indicating whether the script is allowed to execute or not
+     */
+    @Override
+    public boolean isAllowedToExecute(Context context, List<DSpaceCommandLineParameter> commandLineParameters) {
+        return true;
```

Comment:
Following the comment this method should check if the user is authenticated, but you are not checking the `context.getEperson` or something to find out if there is some user in the Context. Is this right?

Reply:
That sort of depends on the caller such as - https://github.com/ufal/clarin-dspace/blob/6d3c3fc46a0a39f79bf702129558deb339d73936/dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ScriptRestRepository.java#L66

There's a bit more detail in https://github.com/ufal/clarin-dspace/pull/1199.

Would `return context.getCurrentUser() != null;` make you more comfortable?

Reply:
Ok, now I can see it. Yes please, that would be cleaner.